src/bt-core-io.c

```diff
@@ -507,31 +506,65 @@ alarm_event(int fd, uint32_t events, void* data)
   return res;
 }
 
-static bool
-set_wake_alarm_cb(uint64_t delay_millis, bool should_wake, alarm_cb cb,
-                  void* data)
+struct set_wake_alarm_param {
+  uint64_t delay_millis;
+  bool should_wake;
+  alarm_cb cb;
+  void* data;
+};
```

Comment:
If possible, I'd prefer to store the exact parameters for |add_relative_timer_| in this structure, move the logic from |set_wake_alarm_task_cb| to |set_wake_alarm_cb|, and keep |set_wake_alarm_task_cb| dumb.